<a href="https://colab.research.google.com/github/satyam-jyoti-sankar/Play_store_data_analysis/blob/main/Copy_of_Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline 

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [12]:
#Here we just read the data frame
df1=pd.read_csv('/content/drive/MyDrive/ Alma Better/ Capstone project-1/Play Store Data.csv')


In [13]:
#it show top 10 data of our 1st data frame
df1.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [14]:
# look at the random 5 records in the apps dataframe
df1.sample(5)

App  ...         Android Ver
1204   Delivery trough - delivery trough delivery trough  ...  Varies with device
4526                                       Q-See QT View  ...          5.0 and up
7150                                          CD Library  ...          2.1 and up
10597                                          Canvas FL  ...          4.1 and up
8900                                         DU Security  ...          4.0 and up

[5 rows x 13 columns]

In [15]:
#Informatation of app dataset
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


# Description of App Dataset columns
App : The name of the app

Category : The category of the app

Rating : The rating of the app in the Play Store

Reviews : The number of reviews of the app

Size : The size of the app

Install : The number of installs of the app

Type : The type of the app (Free/Paid)

The price of the app (0 if it is Free)

Content Rating :The appropiate target audience of the app

Genres: The genre of the app

Last Updated : The date when the app was last updated

Current Ver : The current version of the app

Android Ver : The minimum Android version required to run the app

In [16]:
# type of Category
x=df1['Category'].unique()
print('Total no of category of app present in playstore is:-',len(x))

Total no of category of app present in playstore is:- 34


By diagnosing the data frame, we know that:

There are 13 columns of properties with 10841 rows of data.

Column 'Reviews', 'Size', 'Installs' and 'Price' are in the type of 'object'

Values of column 'Size' are strings representing size in 'M' as Megabytes, 'k' as kilobytes and also 'Varies with devices'.

Values of column 'Installs' are strings representing install amount with symbols such as ',' and '+'.

Values of column 'Price' are strings representing price with symbol '$'.

Hence, we will need to do some data cleaning.

## Data cleaning

In [17]:
#Convert install coloum string to int
def convert_int(x):
    x=x+','+'+'+'M'+'k'+'$'  
    x=x.replace(',','')
    x=x.replace('+','')
    x=x.replace('M','')
    x=x.replace('k','')
    x=x.replace('$','')
    if x=='Free' or x=='Varies with device'or x=='Everyone':
        x=0
    #elif x==
    else:
        x=eval(x)
    
 
    return x


In [18]:
#convert installs,size,price,reviews to int valu so that it better for calcilatation
df1['Installs']=df1['Installs'].apply(convert_int)
 
df1['Size_int']=df1['Size'].apply(convert_int)
df1['Price']=df1['Price'].apply(convert_int)
df1['Reviews']=df1['Reviews'].apply(convert_int)

In [20]:
# Remove wich app reating is more than 5
r=df1[df1['Rating']>5].index 

df1=df1.drop(r)

In [21]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10840 entries, 0 to 10840
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10840 non-null  object 
 1   Category        10840 non-null  object 
 2   Rating          9366 non-null   float64
 3   Reviews         10840 non-null  float64
 4   Size            10840 non-null  object 
 5   Installs        10840 non-null  int64  
 6   Type            10839 non-null  object 
 7   Price           10840 non-null  float64
 8   Content Rating  10840 non-null  object 
 9   Genres          10840 non-null  object 
 10  Last Updated    10840 non-null  object 
 11  Current Ver     10832 non-null  object 
 12  Android Ver     10838 non-null  object 
 13  Size_int        10840 non-null  float64
dtypes: float64(4), int64(1), object(9)
memory usage: 1.6+ MB


In [22]:
# check on null values
df1.isna().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          8
Android Ver          2
Size_int             0
dtype: int64

Here, we realized that there are 1474 rows having null values under column 'Rating'. Hence, we decided to replace the null values with median of overall 'Rating' values.

median=In statistics and probability theory, the median is the value separating the higher half from the lower half of a data sample, a population, or a probability distribution. For a data set, it may be thought of as "the middle" value

In [24]:
x=df1['Rating'].median()
x
df1['Rating']=df1['Rating'].fillna(x)

In [25]:
df1.isna().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              1
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       8
Android Ver       2
Size_int          0
dtype: int64

In [26]:
#Remove the app which has contain no type
y=df1[df1['Type'].isna()].index
df1=df1.drop(index=y)
df1.isnull().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       8
Android Ver       2
Size_int          0
dtype: int64

In [27]:
#Based on higest riview we creat the data set and change the 
df2=pd.DataFrame(df1)
df2.sort_values(by='Reviews',ascending=False, inplace=True)

df2=df2.drop_duplicates(subset=['App'])
df2

App  ... Size_int
2544                                  Facebook  ...      0.0
381                         WhatsApp Messenger  ...      0.0
2604                                 Instagram  ...      0.0
382   Messenger – Text and Video Chat for Free  ...      0.0
1879                            Clash of Clans  ...     98.0
...                                        ...  ...      ...
2440                              Doctors Care  ...     25.0
2439                    Weill Cornell Medicine  ...     19.0
2438                 Pediatric Emergency Guide  ...      3.8
2437                        Dare EMS Protocols  ...     20.0
8019                                  Learn CW  ...     21.0

[9658 rows x 14 columns]

In [28]:
# check on statistical information of the dataframe
df2.describe()

Rating       Reviews      Installs        Price     Size_int
count  9658.000000  9.658000e+03  9.658000e+03  9658.000000  9658.000000
mean      4.192452  2.168266e+05  7.798978e+06     1.097344    32.420553
std       0.496080  1.831524e+06  5.377245e+07    16.852487    96.488804
min       1.000000  0.000000e+00  0.000000e+00     0.000000     0.000000
25%       4.000000  2.500000e+01  1.000000e+03     0.000000     3.400000
50%       4.300000  9.695000e+02  1.000000e+05     0.000000    10.000000
75%       4.500000  2.945775e+04  1.000000e+06     0.000000    28.000000
max       5.000000  7.815831e+07  1.000000e+09   400.000000  1020.000000

# Data visualization